### ISSUE TITLE
bug(all components):  Can't find stylesheet to import.

### ISSUE THREAD

**Issue by nehaGI** [angular/components#29534]
### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

18.0.5

### Description

after upgrading to @angular/material@18.1.2 getting below error on starting local server.

./apps/composer/src/styles.scss - Error: Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
Can't find stylesheet to import.
  ╷
2 │ @use '@material/feature-targeting';
  │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
  node_modules\@ruf\mdc\src\autocomplete\structure.scss 2:1  @forward
  node_modules\@ruf\mdc\src\autocomplete\index.scss 2:1      @forward
  node_modules\@ruf\mdc\index.scss 1:1                       @use
  node_modules\@ruf\shell\card\_card.theme.scss 3:1          @use
  node_modules\@ruf\shell\styles\_all-typography.scss 3:1    @use
  node_modules\@ruf\shell\styles\_all-theme.scss 4:1         @forward
  node_modules\@ruf\shell\index.scss 2:1                     @use
  apps\composer\src\composer-theme.scss 11:1                 @use
  apps\composer\src\styles.scss 2:1                          root stylesheet

This issue is coming because @material/* packages are not getting install in node_modules/@angular folder on version @angular/material version  > 18.1.2

This use to work properly earliar when we are using ~18.0.5 (we areusing m2 components not m3)

### Reproduction

StackBlitz link: 
Steps to reproduce:
1. install latest version of @angular/material@18.1.2 or any greater version.

### Expected Behavior

This should not be giving error. And we should be getting @material/* package get install.

### Actual Behavior

./apps/composer/src/styles.scss - Error: Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
Can't find stylesheet to import.
  ╷
2 │ @use '@material/feature-targeting';
  │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
  node_modules\@ruf\mdc\src\autocomplete\structure.scss 2:1  @forward
  node_modules\@ruf\mdc\src\autocomplete\index.scss 2:1      @forward
  node_modules\@ruf\mdc\index.scss 1:1                       @use
  node_modules\@ruf\shell\card\_card.theme.scss 3:1          @use
  node_modules\@ruf\shell\styles\_all-typography.scss 3:1    @use
  node_modules\@ruf\shell\styles\_all-theme.scss 4:1         @forward
  node_modules\@ruf\shell\index.scss 2:1                     @use
  apps\composer\src\composer-theme.scss 11:1                 @use
  apps\composer\src\styles.scss 2:1     

### Environment

- Angular:
- CDK/Material:
- Browser(s):
- Operating System (e.g. Windows, macOS, Ubuntu):

**Comment by crisbeto** [comment]
In the latest patch version we dropped the dependency on `@material/feature-targeting` so you'll have to install it yourself by adding `material-components-web` to your `package.json`.

**Comment by sandeeppatidar30** [comment]
We are an Angular and Material-based framework team. We styled all Material components according to our style guide and have heavily styled angular material component via material-components-web library.

After the MDC migration, Angular Material used the material-components-web library to style its components. Therefore, we also used material-components-web to style our components, as this library provides many APIs to directly target the Angular Material elements. Now, Angular Material no longer uses it, which broke our approach.

Is there any future-proof approach that can be used to style Material components? Can we use Angular Material CSS variables directly to style these components? Is there any document that we can refer to for all CSS variables of all components? Right now, the only way to find these variables is by using the styles tab of developer tools.
